File: app/src/main/java/vn/loitp/app/activity/customviews/textview/scoretext/ScoreTextViewActivity.java
package vn.loitp.app.activity.customviews.textview.scoretext;

import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.widget.Button;
import android.widget.TextView;

import com.core.base.BaseFontActivity;
import com.core.utilities.LLog;

import vn.loitp.app.R;

public class ScoreTextViewActivity extends BaseFontActivity {
    private TextView tv;
    private Button bt;

    private int maxScore = 200;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        tv = findViewById(R.id.tv);
        bt = findViewById(R.id.bt);

        tv.setText("" + currentScore);

        bt.setOnClickListener(v -> {
            currentScore = 0;
            bt.setEnabled(false);
            updateScore(maxScore);
        });
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_score_text;
    }

    private int currentScore = 0;
    public static final int KEY_UPDATE_SCORE = 1;
    public static final int DELTA_T = 20;//msecond
    Handler handler = new Handler() {
        @Override
        public void handleMessage(Message msg) {
            switch (msg.what) {
                case KEY_UPDATE_SCORE:
                    tv.setText((String) msg.obj);
                    LLog.d(getTAG(), msg.obj + "");
                    if (msg.obj.equals(String.valueOf(maxScore))) {
                        LLog.d(getTAG(), "done");
                        bt.setEnabled(true);
                    }
                    break;
            }
        }
    };

    private void updateScore(int totalScore) {
        for (int i = currentScore; i <= totalScore; i++) {
            handler.sendMessageDelayed(handler.obtainMessage(KEY_UPDATE_SCORE, "" + i), (i - currentScore) * DELTA_T);
        }
    }

}

File: app/src/main/java/vn/loitp/app/activity/customviews/textview/scoreText/ScoreTextViewActivity.kt
package vn.loitp.app.activity.customviews.textview.scoreText

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.os.Message
import com.loitp.annotation.IsFullScreen
import com.loitp.annotation.LogTag
import com.loitp.core.base.BaseFontActivity
import com.loitp.core.ext.setSafeOnClickListener
import com.loitp.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_text_view_score.*
import vn.loitp.R

@LogTag("ScoreTextViewActivity")
@IsFullScreen(false)
class ScoreTextViewActivity : BaseFontActivity() {

    companion object {
        const val KEY_UPDATE_SCORE = 1
        const val DELTA_T = 20 // mls
    }

    private val maxScore = 200

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_text_view_score
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBaseBackPressed()
                }
            )
            this.ivIconRight?.setImageResource(R.color.transparent)
            this.tvTitle?.text = ScoreTextViewActivity::class.java.simpleName
        }
        textView.text = "$currentScore"
        bt.setSafeOnClickListener {
            currentScore = 0
            bt.isEnabled = false
            updateScore(maxScore)
        }
    }

    private var currentScore = 0
    private var handler: Handler = object : Handler(Looper.getMainLooper()) {
        override fun handleMessage(msg: Message) {
            when (msg.what) {
                KEY_UPDATE_SCORE -> {
                    textView.text = msg.obj as String
                    logD(msg.obj.toString())
                    if (msg.obj == maxScore.toString()) {
                        logD("done")
                        bt.isEnabled = true
                    }
                }
            }
        }
    }

    @Suppress("unused")
    private fun updateScore(totalScore: Int) {
        for (i in currentScore..totalScore) {
            handler.sendMessageDelayed(
                handler.obtainMessage(KEY_UPDATE_SCORE, "" + i),
                (i - currentScore) * DELTA_T.toLong()
            )
        }
    }
}
